lazy_theta_star_planner/src/lazy_theta_star2.cpp

```diff
@@ -0,0 +1,257 @@
+//  Copyright 2020 Anshumaan Singh
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http:// www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "lazy_theta_star_planner/lazy_theta_star2.h"
+#include <vector>
+
+namespace lazyThetaStar
+{
+
+LazyThetaStar::LazyThetaStar()
+{
+  sizeX = 0;
+  sizeY = 0;
+}
+
+bool LazyThetaStar::getPath(std::vector<coords<world_pts>> & raw_path)
```

Comment:
I think you need a setGoal and setStart setCostmap functions, you shouldn't be directly accessing the start / goal members directly in the public space. Those should all be private or protected.